### PR TITLE
Test PaymentMethodData data member is converted in constructor

### PR DESCRIPTION
**Pull request by marcoscaceres**
Test for https://github.com/w3c/payment-request/pull/812

### CODE REVIEW

**Review by zouhir — APPROVED** [review]
Lgtm

**Comment by marcoscaceres** [comment]
Thanks @zouhir! If you'll like to check the test in Edge:
https://w3c-test.org/submissions/14205/payment-request/constructor_convert_method_data.https.html

**Comment by zouhir** [comment]
Yep, will do! @marcoscaceres

File: payment-request/constructor_convert_method_data.https.html [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const unknownPMIs = ["fake-pmi", "https://does-not.exist"];
+  promise_test(async t => {
+    for (const supportedMethods of [...knownPMIs, ...unknownPMIs]) {
+      const method = { supportedMethods };
+      const request = new PaymentRequest([method], details);
+      assert_throws(
+        new TypeError(),
+        () => {
+          const badMethod = {
+            ...method,
```
Comment (zouhir) [inline payment-request/constructor_convert_method_data.https.html]: I apologize I oversaw this, spread object properties won't work in Edge. Maybe Object assign?

File: payment-request/constructor_convert_method_data.https.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      const request = new PaymentRequest([method], details);
+
+      // Only check the PMIs that are actually supported
+      if (!(await request.canMakePayment())) continue;
+
+      assert_throws(
+        new TypeError(),
+        () => {
+          const badMethod = {
+            ...method,
```
Comment (zouhir) [inline payment-request/constructor_convert_method_data.https.html]: Same as above

File: payment-request/constructor_convert_method_data.https.html (hunk 1 above)
Comment (marcoscaceres) [reply payment-request/constructor_convert_method_data.https.html]: No probs. Yeah, will do Object.assign.

**Comment by danielramn** [comment]
Bbvbancomer
Clave interbancaria 012180015539350001
ND 0291047225

**Review by zouhir — APPROVED** [review]
works perfect and spread object property issue fixed.

**Pull request merged**